lib/explorer/series.ex

```diff
@@ -2164,6 +2165,51 @@ defmodule Explorer.Series do
 
   def mean(%Series{dtype: dtype}), do: dtype_error("mean/1", dtype, [:integer, :float])
 
+  @doc """
+  Gets the mode value of the series.
+
+  ## Supported dtypes
+
+    * `:integer`
+    * `:float`
+
+  ## Examples
+
+      iex> s = Explorer.Series.from_list([1, 2, 2, 3])
+      iex> Explorer.Series.mode(s)
+      #Explorer.Series<
+        Polars[1]
+        integer [2]
+      >
+
+      iex> s = Explorer.Series.from_list([1, 2, 3, 4])
+      iex> Explorer.Series.mode(s)
+      #Explorer.Series<
+        Polars[4]
+        integer [1, 2, 3, 4]
+      >
+
+      iex> s = Explorer.Series.from_list([1, 1, 2, 2])
+      iex> Explorer.Series.mode(s)
+      #Explorer.Series<
+        Polars[2]
+        integer [1, 2]
+      >
+
+      iex> s = Explorer.Series.from_list(["a", "b", "b", "c"])
+      iex> Explorer.Series.mode(s)
+      #Explorer.Series<
+        Polars[1]
+        string ["b"]
+      >
+  """
+  @doc type: :aggregation
+  @spec mode(series :: Series.t()) :: Series.t() | nil
+  def mode(%Series{dtype: dtype} = series) when is_numeric_or_io_dtype(dtype),
```

Comment:
We use iodtype with other meaning, which would include datetimes, time, etc, so I would call it "is_numeric_or_binary_dtype". Also, would categories work here? They are either integers or strings.

Reply:
Good question! I think they would.

Reply:
Honestly I think it should support any dtype. I'll just have to see what Polars supports and if there's a workaround.

Reply:
Yep it works for all dtypes! 🎉